### ISSUE TITLE
vscode-neovim future & organization

### ISSUE THREAD

**Issue by asvetliakov** [asvetliakov/vscode-neovim#319]
As you probably have noticed, i have been inactive during long time.

Initially this extension was started as fun/POC project - "what if i'd use vscode insert mode capabilities and communicate with neovim for modal editing? Would it be possible?". But it grew into something that I and many others have been using and continue to use every day. Thank you all for your support, suggestions, PRs, really appreciate that.

So, to prevent a stuck state if i'd disappear again (that doesn't mean i will stop support & maintain), i would like to do reorganization here and ask you to participate in discussion how should it look like:

1. Should we move the repo to a new organization, or leave it in the current repo?
2. Anyone want to become an additional maintainer here ?
3. How should we handle donations ? When i enabled them, i really haven't expected that anyone would donate, that was very pleasing, thank you for your support! Btw, in a case if the repo will be maintained by multiple maintainers, i can't feel the right to receive donations anymore. They can be accumulated within org account, transferred to neovim project, or just disabled. 

Let me know what do you think. Thank you

**Comment by Shatur95** [comment]
What if move it to [VSCodeVim](https://github.com/VSCodeVim) organization? This could make the project more popular and attract new contributors.

**Comment by asvetliakov** [comment]
>What if move it to VSCodeVim organization? This could make the project more popular and attract new contributors.

That can be an option. However i'm not sure if VSCodeVim org would want to adopt us 😄 The scopes of projects are different

**Comment by Shatur95** [comment]
Yes, scopes are different but this is great. Your plugin is for Neovim users that want to embed Neovim inside VSCode. Vim plugin is for Vim users that want to emulate Vim behavior without Neovim installation. And the user himself chooses which plugin suits him best.

**Comment by slavoutich** [comment]
> Should we move the repo to a new organization, or leave it in the current repo?

Probably if you find co-maintainers, then it would be useful. Migration is 100% smooth :)

> Anyone want to become an additional maintainer here ?

Don't have any competence in the subj :(

> How should we handle donations?

If they enable you or someone else to spend part of work time on `vscode-neovim`, their goal is achieved. You decide.

**Comment by Shatur95** [comment]
Suggestion: Rename the extension from "Neo Vim" to "Neovim" and add Neovim icon in VSCode marketplace.

**Comment by ngscheurich** [comment]
Hi, @asvetliakov—thanks for your work on this!

> Should we move the repo to a new organization, or leave it in the current repo?

I don't have a strong preference here, but I do feel that the name should be changed. I'd recommend renaming the project to something like "NeoCode" so that it's distinguishable from the Neovim project (e.g. no special context is needed). I'd also recommend adding a logo of some kind, perhaps that incorporates elements of VS Code and Neovim, so that it looks more legitimate in the marketplace. I'd be willing to build a logo if you're interested.

> Anyone want to become an additional maintainer here ?

I am interested in contributing, as Vim/Neovim with a modern frontend is quite appealing to me. I can't take on the burden of being a full- or even co-maintainer, but if we can get together a group of people, I'd like to be part of it.

> How should we handle donations ?

If the project is already receiving non-trivial donations, I'd be for passing all of those on to the Neovim project or a worthwhile charity.

**Comment by ngscheurich** [comment]
> I'd also recommend adding a logo of some kind, perhaps that incorporates elements of VS Code and Neovim

**Comment by ngscheurich** [comment]
> I'd also recommend adding a logo of some kind, perhaps that incorporates elements of VS Code and Neovim

I added a proposal that we can perhaps iterate on in #325.

**Comment by asvetliakov** [comment]
>Suggestion: Rename the extension from "Neo Vim" to "Neovim" (this is how it written on the official website) and add Neovim icon in VSCode marketplace.

I don't think it's right to use another branding. Although it's scoped to vscode, but still can be confusing to users and especially search systems.

>NeoCode

👍

>I added a proposal that we can perhaps iterate on in #325.

Awesome!

**Comment by ftassi** [comment]
@asvetliakov thank you for your work so far and also for raising this issue about the future of the project. I'm not sure how many co-maintainer showed up since you opened this issue, but I think that the most crucial matter here is involving as much people as possible in the development of the project. Have you thought about labeling some issue with "easy pick" to ease new developers into the project? It could help a bit. 

I'm migrating to vscode / noevim right now, so I'm not really aware of the state of this repo yet, so the following question may sound a bit silly, but still.... How many maintainers are there, other than you, right now? Is this still a "one man band" project :wink: ?

**Comment by colinxs** [comment]
@asvetliakov You could consider seeing if the Neovim devs would be open to moving vscode-neovim to the Neovim org. It looks like theres a Visual Studio (not Code) extension there already: https://github.com/neovim/VSNvim.

Now that Onivim has abandoned Neovim this could be a good way for Neovim to showcase its use as a headless backend :)

**Comment by justinmk** [comment]
I have been developing a vscode extension for work, so I may be able to help here.

**Comment by asvetliakov** [comment]
@ftassi Only me primary. There are few active people here who also help/send PR too. `Easy pick/Good first issue` is good suggestion 👍 

@colinxs @justinmk I don't mind moving it to neovim org 💯  . Will also greatly increase the visibility and improve API/features feedback/communication.

**Comment by DAlperin** [comment]
A bit late to this thread but I would definitely be happy to help out around here! I have a local fork with a bunch of bug fixes and such that when I have a free moment from work I would love to upstream at some point.

**Comment by asvetliakov** [comment]
@DAlperin 
you definitely should look into #334 before making a PR. A lot of things have changed there

**Comment by DAlperin** [comment]
Oh! Very cool. Looks like some of my changes still apply but I will definitely have to rework them a bit.

**Comment by Shatur95** [comment]
Suggestion: provide [issue templates](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository) to improve issues quality and ask user in this templates to search for similar issue first.

**Comment by Nico-Guyon** [comment]
A bit late also but I would like to get involved on this project.

I was using vscodevim before switching to vscode neovim which I find much powerful and fast.
However lately an annoying bug where neovim can't find the windows id of the editor keeps appearing.
It's annoying me enough to make me want to fix this one, so I forked the project and started looking into it.

This extension became part of my everyday work and I'd like it to stay that way.
Is there a list of all maintainers somewhere, how this project is organized ?

**Comment by Shatur95** [comment]
> It's annoying me enough to make me want to fix this one, so I forked the project and started looking into it.

Looking forward for the fix :)

Currently @theol0403 and me can merge PR's. But we are missing TS developers.

**Comment by theol0403** [comment]
Yeah, after we both contributed to this repo the owner added us as collaborators to help manage PRs. However, he is still in charge of development, it's just that his time to work on this project is intermittent as far as I can tell. PRs and bug reports are very appreciated, and from what I've seen he does have the time to comment on significant PRs. But yeah we are just missing core TS developers.

Also, while much can definitely be done to improve this plugin, like fixing bugs and improving features, significant performance and usability improvements are more often than not being blocked by vscode devs, like the notorious lag that is caused from vscode plugin host being singlethreaded which makes neovim in insert mode impractical (even though in theory it can totally be done).

**Comment by likern** [comment]
@asvetliakov Man, this is an amazing extension! So smooth and awesome working with VSCode. I only started exploring again Vim, after abandoning it many years ago, but it looks like the best possible experience out there.

And ... it just works almost like native NeoVim. I will donate soon.

But if you, with other contributors, could provide commercial, on top of open-source, license to increase speed of development, better bugfixing, etc - I'm totally in. I think it has big commercial potential. I would compare it as quokka (which looks like has a sustainable profit).

I think this is the best and only real sustainable model to support velocity of development.
Even if you are doing open-source for fun, or using it yourself, a lot of chanches that sooner or later it will be abandoned. Like many many open-source project, done as hobby.

**Comment by asvetliakov** [comment]
>But if you, with other contributors, could provide commercial, on top of open-source, license to increase speed of development, better bugfixing, etc - I'm totally in. I think it has big commercial potential. I would compare it as quokka (which looks like has a sustainable profit).

Won't happen. This is fun project and I never considered it as possibly profit-generating. Another thing i don't want to deal with paid support/feature requests which will force me to do something i don't want, impossible to do, or will take ridiculous amount of time to implement. Nvim & vscode interop is very tricky and has a lot of compromises, which people doesn't always like or willing to accept. If you're eager to support me you can buy me or any active contributor a ☕  or 🍷  or so 😄  . But for premium, company-like support you should look into other projects, for example [Onivim2](https://onivim.io/) (if it'd be ever finished)

**Comment by likern** [comment]
> Won't happen. This is fun project and I never considered it as possibly profit-generating.

Totally understand. Hope you will not abandon it.
> If you're eager to support me you can buy me or any active contributor a coffee or wine_glass or so smile . But for premium, company-like support you should look into other projects, for example Onivim2 (if it'd be ever finished)

Done.

I carefully checked out their goals and plans. Looks too ambitous to me. As I understand they develop their own editor, not based on VSCode source code, using Reason language and their own UI framework. At the same time trying to support full VSCode extensions API. I wish them all the best and hope they could reach their goals. But, frankly speaking, I don't believe into this specific technological stack.

**Comment by asvetliakov** [comment]
>Totally understand. Hope you will not abandon it.

Don't worry. I may be inactive from time to time, but i'm using vscode & this extension every day. So unless i'd get crazy idea to stop programming, quit my job and become a farmer man 👨🏻‍🌾  🚜  🍇  or something like it, i will continue to support it

**Comment by MaximoMachado** [comment]
Just wanted to say, thanks so much for this wonderful extension. It blew my mind that something like this is actually possible and I've been telling everyone and everything about it.

I'd love to be able to contribute to the project in a meaningful way but I'm not sure of a good way to get started. To be frank, I haven't worked with VSCode extensions at all in the past but hopefully my enthusiasm makes up for the inexperience.

How would you recommend getting started contributing to the project? Any pointers would be incredibly helpful.

**Comment by trkoch** [comment]
> You could consider seeing if the Neovim devs would be open to moving vscode-neovim to the Neovim org.

@colinxs I really like that idea. Its also a good companion to `node-client` which is already part of the Neovim organization and used by this extension.

>> What if move it to VSCodeVim organization? This could make the project more popular and attract new contributors.

> That can be an option. However i'm not sure if VSCodeVim org would want to adopt us 😄 The scopes of projects are different

Perhaps they are willing to add a link to your project in the [README](https://github.com/VSCodeVim/Vim/blob/master/README.md)? I'd hope they don't see this project as harmful competition, but rather as incitement. Also, browsing through Vim plugins, there is a nice tradition of mentioning alternatives.

**Comment by samvv** [comment]
Do you still need more maintainers? I might be interested. I have developed a few VSCode extensions and I've also integrated Neovim into Electron a while back, so I'm quite used to the API. I'm using this extension on a daily basis so there's certainly motivation to keep this extension alive and well :)

**Comment by Shatur95** [comment]
@samvv, yes, it would be awesome.

**Comment by asvetliakov** [comment]
@samvv I've sent you the invite. Welcome! 

PS. I'm little inactive again, sorry

**Comment by samvv** [comment]
Oh wow thanks! Happy to help!
-------- Original Message --------

**Comment by samvv** [comment]
Going to use this afternoon to work a bit on this repository. Hopefully I'll be able to close a few issues.

**Comment by trkoch** [comment]
@samvv Great to see other contributors!

**Comment by samvv** [comment]
Thanks 😄 Its difficult to combine this with building a web programming language, but I'll try to have a look at some issues this weekend.

**Comment by trkoch** [comment]
Would anybody be interested in joining a platform for group/direct communication, e.g. a Gitter channel or some kind of forum? Personally I have some questions (setup, design) which are probably best asked at a place like this.

**Comment by asvetliakov** [comment]
I've just created [gitter community](https://gitter.im/vscode-neovim/community?utm_source=share-link&utm_medium=link&utm_campaign=share-link): feel free to join

**Comment by ar1a** [comment]
> I've just created [gitter community](https://gitter.im/vscode-neovim/community?utm_source=share-link&utm_medium=link&utm_campaign=share-link): feel free to join

I feel like this sort of thing would be better suited to discord, but maybe i'm spoiled from doom emacs 😅

**Comment by yogeshdhamija** [comment]
I'd also like to contribute. I have some experience with TS, and this is my daily driver for work.

**Comment by yogeshdhamija** [comment]
> I'd also like to contribute. I have some experience with TS, and this is my daily driver for work.

In hindsight, it was pretty arrogant of me to offer to help without any meaningful portfolio attached to my public profile. I'm happy to discuss my qualifications privately, or to let it lie.

Regardless, it seems there hasn't been much movement in this repo for quite a while. I've been waiting on a PR (#723) with bated breath that seems to be trivial to merge or update. Is there any movement on how the maintenance of this project should be handled moving forward? 

This is a wonderful tool and I'm absurdly grateful to @asvetliakov and all the maintainers. It's become my daily driver and it gives me anxiety to imagine a world where it goes unmaintained.

**Comment by abosnjakovic** [comment]
I float between this and my neovim config and I find it very useful, I'd like to help maintain this if you are looking for contributors.

**Comment by trkoch** [comment]
@theol0403 Thanks for taking care of the PRs! I was running on a separate version and believe I can now move back to main. Still I noticed that releases happen rarely (please don't understand it as criticism, it's meant only as a statement of facts). Have you thought about [pre-releases](https://code.visualstudio.com/updates/v1_63#_pre-release-extensions)? This would allow interested users to test the latest and greatest, without sacrificing the stability that comes with infrequent releases.

**Comment by theol0403** [comment]
Np @trkoch. Still haven't merged everything in yet, but it's a start.

Unfortunately, I do not have the ability to commit to main, merge my own PRs, or make releases. If @asvetliakov wants to grant me those, I would be willing to make releases. I don't really think prereleases are necessary, with the experimental nature of this project (and the numerous bugs), it would probably be better to roll out fixes as soon as they are merged.

**Comment by trkoch** [comment]
@asvetliakov Thank you for starting this thread. Quite a few people have offered support. Recently, there have been some PRs with popular improvements. I guess we need some kind of suggestion from your side to find a way forward.

**Comment by theol0403** [comment]
Hey all, I just finished up my finals at school and will have some time in the next coming weeks to give this repo some much-needed attention - merge PRs, triage issues, appoint contributors, and hopefully contribute myself. 

Also, it looks like @justinmk is getting involved - this is amazing news! One thing I've been wondering about is refactoring the plugin to better use the nvim lua api rather than the hacks it does for stuff like file saving. Also maybe advancements in the neovim api in the last year can solve some major issues. In any case, I'll do my best to make it possible for more people to be involved and contribute.

**Comment by justinmk** [comment]
> Also, it looks like @justinmk is getting involved - this is amazing news!

:) I'm doing what I can but won't have much time. I also don't have Admin rights so I can't do things like publish a release (which is badly needed).

> One thing I've been wondering about is refactoring the plugin to better use the nvim lua api rather than the hacks it does for stuff like file saving

Nvim API improvements could afford a ton of simplifications which may also lead to more stability. My suggestions for this project:

1. Require latest Nvim release. With the current maintenance pace, it doesn't make sense to spend energy supporting old versions. VSCode itself updates aggressively, so requiring an updated Nvim is reasonable.
1. _Deleting_ code could be the theme of this project for awhile. There are a lot of "nice to have" features that add risk. Limitations are good.
1. CI should produce VSIX artifacts. Then users at least have a chance of updating, even if the marketplace version is old.

**Comment by theol0403** [comment]
> :) I'm doing what I can but won't have much time. I also don't have Admin rights so I can't do things like publish a release (which is badly needed).

Great points! Although I understand little of nvim and the business logic of the plugin (yet), it definitely seems like modernizing this plugin will have many advantages.

I've added you as admin. I think you need credentials to publish a release, however. I have them, and I'll try to make a release in the next couple of days.

**Comment by trkoch** [comment]
I just realized it's _the_ "sneak" @justinmk 😀 . Thank you for your contribution to the vim ecosystem and very much welcome onboard!

**Comment by trkoch** [comment]
> Would anybody be interested in joining a platform for group/direct communication, e.g. a Gitter channel or some kind of forum?

On second thought, GitHub discussion may be more appropriate considering discussions here typically span days or weeks rather than minutes or hours.

> Deleting code could be the theme of this project for awhile. There are a lot of "nice to have" features that add risk. Limitations are good.

This would also be a great place to discuss initiative or priorities like those.

**Comment by macintacos** [comment]
My $0.02: use GitHub Discussions. It's integrated, and getting a bunch of attention from GitHub anyway, so it's likely to only get better over time. Plus, it keeps this repo as the "hub" of activity.

**Comment by trkoch** [comment]
Anybody object converting this issue to a discussion? It can be pinned there just the same.
